microlens-th-0.4.3.7 fails to build with th-abstraction-0.3.*

When attempting to build [`llvm-pretty-0.11.0`](http://hackage.haskell.org/package/llvm-pretty-0.11.0) (which requires `th-abstraction-0.3.1` or later), I stumbled over a build failure in `microlens-th`:

```
Configuring library for microlens-th-0.4.3.7..
Preprocessing library for microlens-th-0.4.3.7..
Building library for microlens-th-0.4.3.7..
[1 of 2] Compiling Lens.Micro.TH.Internal ( src/Lens/Micro/TH/Internal.hs, dist/build/Lens/Micro/TH/Internal.o, dist/build/Lens/Micro/TH/Internal.dyn_o )

src/Lens/Micro/TH/Internal.hs:57:1: error:
    Could not load module ‘Language.Haskell.TH.Datatype.TyVarBndr’
    It is a member of the hidden package ‘th-abstraction-0.4.0.0’.
    Perhaps you need to add ‘th-abstraction’ to the build-depends in your .cabal file.
    Use -v (or `:set -v` in ghci) to see a list of the files searched for.
   |
57 | import           Language.Haskell.TH.Datatype.TyVarBndr
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Indeed, `Language.Haskell.TH.Datatype.TyVarBndr` wasn't introduced until `th-abstraction-0.4.0.0`, so I think `microlens-th`'s lower version bounds on `th-abstraction` should be `>= 0.4`, not `>= 0.2.1`.

--- a/microlens-th/microlens-th.cabal
+++ b/microlens-th/microlens-th.cabal
@@ -42,7 +42,7 @@ library
                      , transformers
                      -- lens has >=2.4, but GHC 7.4 shipped with 2.7
                      , template-haskell >=2.7 && <2.18
-                     , th-abstraction >=0.2.1 && <0.5
+                     , th-abstraction >=0.4 && <0.5
 
   ghc-options:
     -Wall -fwarn-tabs

